ctk/idp/src/main/kotlin/org/codice/compliance/web/sso/PostSSOTest.kt

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.compliance.web.sso
+
+import com.jayway.restassured.RestAssured
+import com.jayway.restassured.RestAssured.given
+import com.jayway.restassured.response.Response
+import de.jupf.staticlog.Log
+import io.kotlintest.specs.StringSpec
+import org.apache.wss4j.common.saml.builder.SAML2Constants
+import org.codice.compliance.Common
+import org.codice.compliance.debugPrettyPrintXml
+import org.codice.compliance.debugWithSupplier
+import org.codice.compliance.saml.plugin.IdpSSOResponder
+import org.codice.compliance.utils.TestCommon.Companion.AUTHN_REQUEST
+import org.codice.compliance.utils.TestCommon.Companion.EXAMPLE_RELAY_STATE
+import org.codice.compliance.utils.TestCommon.Companion.ID
+import org.codice.compliance.utils.TestCommon.Companion.SP_ISSUER
+import org.codice.compliance.utils.TestCommon.Companion.acsUrl
+import org.codice.compliance.utils.TestCommon.Companion.authnRequestToString
+import org.codice.compliance.utils.TestCommon.Companion.getServiceProvider
+import org.codice.compliance.utils.decorate
+import org.codice.compliance.verification.binding.BindingVerifier
+import org.codice.compliance.verification.core.responses.AuthnRequestProtocolResponseVerifier
+import org.codice.compliance.verification.profile.SingleSignOnProfileVerifier
+import org.codice.security.saml.SamlProtocol.Binding.HTTP_POST
+import org.codice.security.saml.SamlProtocol.POST_BINDING
+import org.codice.security.sign.Encoder
+import org.codice.security.sign.SimpleSign
+import org.joda.time.DateTime
+import org.opensaml.saml.common.SAMLVersion
+import org.opensaml.saml.saml2.core.impl.AuthnRequestBuilder
+import org.opensaml.saml.saml2.core.impl.IssuerBuilder
+import org.opensaml.saml.saml2.core.impl.NameIDPolicyBuilder
+
+class PostSSOTest : StringSpec() {
+    companion object {
+
+        /** Sets up positive path tests.
+         * @return A string representation of a valid encoded POST AuthnRequest.
+         */
+        private fun createValidAuthnRequest(): String {
```

Comment:
❓ Should we extract `createValidAuthnRequest` to `TestCommon`? One for Post and one for Redirect

Reply:
I will do this in a follow-on PR where I have to rework some of these utility functions anyways.